src/SonarScanner.MSBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -88,16 +83,16 @@ private string TryWriteSonarLintXmlFile(IEnumerable<SonarRule> activeRules)
         }
         else
         {
-            var content = RoslynSonarLint.GenerateXml(activeRules, sonarProperties, language);
+            var content = RoslynSonarLint.GenerateXml(rules.Where(x => x.IsActive), sonarProperties, language);
             logger.LogDebug(Resources.RAP_WritingAdditionalFile, sonarLintXmlPath);
             File.WriteAllText(sonarLintXmlPath, content);
             return sonarLintXmlPath;
         }
     }
 
-    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins(IEnumerable<SonarRule> activeRules)
+    private IEnumerable<AnalyzerPlugin> FetchAnalyzerPlugins()
     {
-        var partialRepoKeys = ActiveRulesPartialRepoKeys(activeRules);
+        var partialRepoKeys = ActiveRulesPartialRepoKeys(rules.Where(x => x.IsActive));
```

Comment:
We give it active, it does not give us result for active-only
```suggestion
        var partialRepoKeys = PartialRepoKeys(rules.Where(x => x.IsActive));
```